index.js

```diff
@@ -11,16 +11,56 @@ if (process.env.NODE_ENV !== 'production') {
     Symbol.for('react.element')) ||
     0xeac7;
 
+  var REACT_PROVIDER_TYPE =
+    (typeof Symbol === "function" &&
+      Symbol.for &&
+      Symbol.for("react.provider")) ||
+    0xeacd;
+
+  var REACT_CONTEXT_TYPE =
+    (typeof Symbol === "function" &&
+      Symbol.for &&
+      Symbol.for("react.context")) ||
+    0xeace;
+
+  var REACT_FORWARD_REF_TYPE =
+    (typeof Symbol === "function" &&
+      Symbol.for &&
+      Symbol.for("react.forward_ref")) ||
+    0xead0;
+
   var isValidElement = function(object) {
     return typeof object === 'object' &&
       object !== null &&
       object.$$typeof === REACT_ELEMENT_TYPE;
   };
 
+  var isValidComponent = function(component) {
+    // Stateless or stateful component
+    if (typeof component === 'function') {
+      return true;
```

Comment:
this seems like it would return a lot of false positives, but i don't know of a better way to cover SFCs. It seems like an early `return true` for class-based components, though, would be better. You may want to use the `react-is` package here, and check the various "is" methods contained within.

Reply:
I agree, it was my first though but went for a first version before digging into this. If it's a function (not a class), it should accept it. If it's a class, we should check if it inherits `React.Component` or from `React.createClass`. Not sure if I can do this with enough reliability.